tests/memory_storage.rs

```diff
@@ -24,3 +24,98 @@ impl Tester<Key, MemoryStorage> for MemoryTester {
 }
 
 generate_store_tests!(tokio::test, MemoryTester);
+
+#[cfg(any(feature = "alter-table", feature = "index", feature = "transaction"))]
+macro_rules! exec {
+    ($glue: ident $sql: literal) => {
+        $glue.execute($sql).unwrap();
+    };
+}
+
+#[cfg(any(feature = "alter-table", feature = "index", feature = "transaction"))]
+macro_rules! test {
+    ($glue: ident $sql: literal, $result: expr) => {
+        assert_eq!($glue.execute($sql), $result);
+    };
+}
+
+#[cfg(feature = "alter-table")]
+#[test]
+fn memory_storage_alter_table() {
+    use gluesql::{Error, Glue};
+
+    let storage = MemoryStorage::default();
+    let mut glue = Glue::new(storage);
+
+    exec!(glue "CREATE TABLE Footer (id INTEGER);");
+    test!(
+        glue "ALTER TABLE Footer RENAME TO Header;",
+        Err(Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()))
+    );
+    test!(
+        glue "ALTER TABLE Footer RENAME COLUMN id TO od;",
+        Err(Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()))
+    );
+    test!(
+        glue "ALTER TABLE Footer ADD COLUMN ratio FLOAT DEFAULT 1.0;",
+        Err(Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()))
+    );
+    test!(
+        glue "ALTER TABLE Footer DROP COLUMN id;",
+        Err(Error::StorageMsg("[MemoryStorage] alter-table is not supported".to_owned()))
+    );
+}
+
+#[cfg(feature = "index")]
+#[test]
+fn memory_storage_index() {
+    use futures::executor::block_on;
+    use gluesql::{
+        store::{Index, Store},
+        Error, Glue, Result,
+    };
+
+    let storage = MemoryStorage::default();
+
+    assert_eq!(
+        block_on(storage.scan_data("Idx"))
+            .unwrap()
+            .collect::<Result<Vec<_>>>()
+            .as_ref()
+            .map(Vec::len),
+        Ok(0),
+    );
+
+    assert_eq!(
+        block_on(storage.scan_indexed_data("Idx", "hello", None, None)).map(|_| ()),
+        Err(Error::StorageMsg(
+            "[MemoryStorage] index is not supported".to_owned()
+        ))
```

Comment:
```suggestion
        block_on(storage.scan_indexed_data("Idx", "hello", None, None)).unwrap_err(),
        Error::StorageMsg(
            "[MemoryStorage] index is not supported".to_owned()
        )
```
how about this change?

Reply:
If you change it, you have to change the other conditions as well.

Reply:
The content `Ok` of `storage.scan_indexed_data` is `Iterator` and it does not implement `Debug`.
`unwrap_err` requires `Debug` impl for the `Ok(T)`, so I simply replaced `Iterator` to `()`.

Reply:
oh, need a debug implementation
It's good to leave it like this.